sdv/single_table/base.py

```diff
@@ -270,6 +270,41 @@ def get_metadata(self):
         """Return the ``SingleTableMetadata`` for this synthesizer."""
         return self.metadata
 
+    def auto_assign_transformers(self, data):
+        """Automatically assign the required transformers for the given data and constraints.
+
+        This method will automatically set a configuration to the ``rdt.HyperTransformer``
+        with the required transformers for the current data.
+
+        Args:
+            data (pandas.DataFrame):
+                The raw data (before any transformations) that will be used to fit the model.
+        """
+        constrained = self._data_processor._fit_transform_constraints(data)
+        columns_created_by_constraints = set(constrained.columns) - set(data.columns)
+        self._data_processor._fit_numerical_formatters(data)
+        config = self._data_processor._create_config(constrained, columns_created_by_constraints)
+        self._data_processor._hyper_transformer.set_config(config)
```

Comment:
These steps are very similar to what happens in the `DataProcessor.fit`. Should we make another method that sets up the hypertransformer as well as does the constraints and formatting in the `DataProcessor` and then call that method both here and in `DataProcessor.fit`?